rosetta-source/src/main/rosetta/legaldocumentation-master-type.rosetta

```diff
@@ -46,13 +47,93 @@ type MasterAgreementVariableSet: <"Defines a type where additional variables ass
-	equityCashSettlementDates PaymentDates (1..1) <"The parameters used to generate the payment date schedule, relative to the equityCalculationPeriod. Part 1 Section 12, 'Definitions', of the 2018 ISDA CDM Equity Confirmation. Para 73: 'Equity Cash Settlement Date' means each date falling one Settlement Cycle after an Equity Valuation Date; provided that if any such date is not a Settlement Currency Business Day, then such date shall be adjusted per Following Day Adjustment.">
+    typeOfSwapElection ReturnTypeEnum (1..1) <"Per Part 1 Section 4, 'Dividend Obligations', of the 2018 ISDA CDM Equity Confirmation, Para 4.2 'Dividend Returns'">
+    pricingMethodElection PriceReturnTerms (1..1) <"Per Part 1 Section 5, 'Pricing', of the 2018 ISDA CDM Equity Confirmation, Para 5.1">
+    linearInterpolationElection InterpolationMethodEnum (1..1) <"Per Part 1 Section 3, 'Floating Obligations', of the 2018 ISDA CDM Equity Confirmation. Para 3.3">
+    settlementTerms SettlementTerms (1..1) <"Per Part 1 Section 8, 'Settlement', of the 2018 ISDA CDM Equity Confirmation for Security Equity Swap">
+    valuationDates ValuationDates (1..1) <"The parameters used to generate the 'Equity Valuation Dates' schedule, including the Effective Date and Termination Date for the Swap.">
+    equityCashSettlementDates PaymentDates (1..1) <"The parameters used to generate the payment date schedule, relative to the equityCalculationPeriod. Per Part 1 Section 12, 'Definitions', of the 2018 ISDA CDM Equity Confirmation. Para 73">
 
 type EquityMasterConfirmation extends MasterConfirmationBase: <"Specification for General Terms and Elections of an Equity Master Confirmation that is applicable across multiple Equity confirmations and is referenced by each of these confirmations, an example of which being the 2018 ISDA CDM Equity Confirmation for Security Equity Swap.">
 
 type MasterConfirmationBase: <"Legal agreement specification for General Terms and Elections that are applicable across multiple confirmations and are referenced by these confirmations.">
+
+type TransactionAdditionalTerms: <"Additional specification for the extraordinary events that may affect a trade and the related contractual rights and obligation of the parties when this happens. Such terms are typically required to extend the economics terms, for the purpose of producing the final legal contractual form of the Transaction.">
+    equityAdditionalTerms EquityAdditionalTerms (0..1)
+    foreignExchangeAdditionalTerms FxAdditionalTerms (0..1)
+    commoditiesAdditionalTerms string (0..1)
+    creditAdditionalTerms string (0..1)
+    interestRateAdditionalTerms string (0..1)
+    digitalAssetAdditionalTerms string (0..1)
+
+type EquityAdditionalTerms: <"Transaction AdditionalTerms that apply to Equity asset class.">
+    extraordinaryEvents ExtraordinaryEvents (0..1)
+    determinationTerms DeterminationRolesAndTerms (1..*)
```

Comment:
We have "determinationTerms" as mandatory, but the new type "DeterminationRolesAndTerms" is an empty type. Ruddy and I have discussed this and agreed that this will be addressed in the next PR in this series (there are 3 planned for Extraordinary events). At the moment the JSON will need to specify this typebut it would be an empty array i.e. "determinationTerms": [ { } ]